README.md

```diff
@@ -0,0 +1,338 @@
+
+- The entire codebase is written in **TypeScript**, to ensure application type safety and prevent bugs. 
+
+
+###Front-end: 
+<img src='/images/reactjs.svg' alt='React Icon' width=48 height=48/>
+<img src='/images/css.svg' alt='CSS Icon' width=48 height=48/>
+<img src='/images/reactbootstrap.svg' alt='React Bootstrap Icon' width=48 height=48/>
+
+- Interactive home page and details page are built with **React.js**, connected together with **React Router**
+- Components are styled with combination of **CSS** and **React-Bootstrap**
+
+###API:
+<img src='/images/nodejs-vertical-dark.svg' alt='Node.js Icon' height=48/>
+<img src='/images/expressjs-ar21.svg' alt='Express.js Icon' height=48/>
+<img src='/images/jest.svg' alt='Jest Icon' height=48/>
+
+- API is constructed with **Node.js** + **Express.js**, implementing routes, middlewares, controllers, and services
+- Unit-testing is integrated into API, carried out with **Supertest** and **Jest**. Tests every API endpoint extensively (i.e., POST cards, GET card ratings, etc...)
+
+###Database: 
+<img src='/images/mongodb2-horizontal.svg' alt='Jest Icon' height=48/>
+<img src='/images/mongoose-logo.png' alt='Jest Icon' height=48/>
+
+- All card information (i.e., price histories, card ratings, etc) stored in database created with **MongoDB**
+- Using **Mongoose** (a MongoDB object modeling tool) to standardize database queries with models/schemas
+
+
+
+---
+## Setup
+You can run this web-app on your local environment in three simple steps: 
+
+###1. Pre-steps
+- Install npm (or yarn) and Node.js
+- Clone the repository onto your local machine
+- Navigate to `/api` directory and install dependencies: `npm install` OR `yarn install` 
+- Create .env files: `.env.development.local`, `.env.production.local`, `.env.test.local` with the following code: 
+```
+LOG_DIR=logs
+PORT=3000
+DB_HOST=localhost
+DB_PORT=27017
+DB_DATABASE=pokemon
+```
+- Navigate to `/app` directory and install dependencies: `npm install` OR `yarn install` 
+
+
+###2. Run the API
+
+- Navigate to the `/api` directory. 
+- Run the API by using: `npm run start` OR `yarn start`
+- This will start the API on port 3000
+
+
+###3. Input your own cards
+
+- Install the [MongoDB Compass GUI](https://www.mongodb.com/try/download/compass)
+- Create a JSON file that contains all of your cards (See `/app/src/realCards.json` for an example of the file format). 
+- Open MongoDB Compass and create new connection to connect to a MongoDB deployment. 
+- On the left hand side, click on the `pokemon` database, then click on the `pokemoncards` folder. 
+- Add data into this folder by importing your JSON file containing your cards
+
+
+###4. Start up the React app
+
+- Navigate to the `/app` directory. 
+- Run the React app by using: `npm start` OR `yarn start`
+- This will start the app on port 3001
+
+You can now effectively manage and track your Pokémon card collection! 
+
+---
+## File Structure
+
+```
+├── api
```

Comment:
I was trying to research ways to do this automatically. Did you do it manually?

Reply:
I was actually contemplating doing it manually haha, but I found this command called `tree` that analyzes your directory and creates a nice file structure for it: https://www.geeksforgeeks.org/tree-command-unixlinux/#

I used a custom command: `tree -I 'coverage|dist|node_modules' --prune > file_structure.txt` (courtesy of gpt lol)

This removed all directories named coverage, dist, and node_modules from the tree (otherwise, it would've been 70,000 lines long literally) and fed the result into a .txt file.